src/game/client.cpp

```diff
@@ -3240,142 +3240,29 @@ namespace client
         }
 
         if(serversortstyles.empty()) updateserversort();
         loopv(serversortstyles)
         {
             int style = serversortstyles[i];
-            serverinfo *aa = a, *ab = b;
             bool reverse = false;
             if(style < 0)
             {
                 style = 0-style;
                 reverse = true;
             }
 
-            switch(style)
-            {
-                case SINFO_STATUS:
-                {
-                    retsw(serverstat(aa), serverstat(ab), true);
-                    break;
-                }
-                case SINFO_DESC:
-                {
-                    retcp(strcmp(aa->sdesc, ab->sdesc));
-                    break;
-                }
-                case SINFO_MODE:
-                {
-                    if(aa->attr.length() > 1) ac = aa->attr[1];
-                    else ac = 0;
-
-                    if(ab->attr.length() > 1) bc = ab->attr[1];
-                    else bc = 0;
-
-                    retsw(ac, bc, true);
-                }
-                case SINFO_MUTS:
-                {
-                    if(aa->attr.length() > 2) ac = aa->attr[2];
-                    else ac = 0;
-
-                    if(ab->attr.length() > 2) bc = ab->attr[2];
-                    else bc = 0;
-
-                    retsw(ac, bc, true);
-                    break;
-                }
-                case SINFO_MAP:
-                {
-                    retcp(strcmp(aa->map, ab->map));
-                    break;
-                }
-                case SINFO_TIME:
-                {
-                    if(aa->attr.length() > 3) ac = aa->attr[3];
-                    else ac = 0;
-
-                    if(ab->attr.length() > 3) bc = ab->attr[3];
-                    else bc = 0;
-
-                    retsw(ac, bc, false);
-                    break;
-                }
-                case SINFO_NUMPLRS:
-                {
-                    retsw(aa->numplayers, ab->numplayers, false);
-                    break;
-                }
-                case SINFO_MAXPLRS:
-                {
-                    if(aa->attr.length() > 4) ac = aa->attr[4];
-                    else ac = 0;
-
-                    if(ab->attr.length() > 4) bc = ab->attr[4];
-                    else bc = 0;
-
-                    retsw(ac, bc, false);
-                    break;
-                }
-                case SINFO_PING:
-                {
-                    retsw(aa->ping, ab->ping, true);
-                    break;
-                }
-                case SINFO_PRIO:
-                {
-                    retsw(aa->priority, ab->priority, false);
-                    break;
-                }
-                default: break;
-            }
+            comp = a->compare(*b, style, reverse);
+            if( comp != 0 ) return comp < 0;
         }
-        return strcmp(a->name, b->name);
+        return strcmp( a->name(), b->name() ) < 0;
```

Comment:
Meaning changed in comparison to old code (should likely be `!= 0`, right?). If this is *really* intentional, then there should be a comment explaining why you consider all values `>= 0` here.